src/lib.rs

```diff
@@ -116,1087 +116,23 @@ pub mod abstraction;
-    pub fn tr_sess_set_attributes(
-        &mut self,
-        handle: ESYS_TR,
-        attributes: TpmaSession,
-    ) -> Result<()> {
-        let ret = unsafe {
-            Esys_TRSess_SetAttributes(
-                self.mut_context(),
-                handle,
-                attributes.flags(),
-                attributes.mask(),
-            )
-        };
-        let ret = Error::from_tss_rc(ret);
-        if ret.is_success() {
-            Ok(())
-        } else {
-            Err(ret)
-        }
-    }
-
-    /// Get session attribute flags.
-    pub fn tr_sess_get_attributes(&mut self, handle: ESYS_TR) -> Result<TpmaSession> {
-        let mut flags: TPMA_SESSION = 0;
-        let ret = unsafe { Esys_TRSess_GetAttributes(self.mut_context(), handle, &mut flags) };
-        let ret = Error::from_tss_rc(ret);
-        if ret.is_success() {
-            Ok(TpmaSessionBuilder::new().with_flag(flags).build())
-        } else {
-            Err(ret)
-        }
-    }
-
-    ///////////////////////////////////////////////////////////////////////////
-    /// Private Methods Section
-    ///////////////////////////////////////////////////////////////////////////
-
-    /// Returns a mutable reference to the native ESYS context handle.
-    fn mut_context(&mut self) -> *mut ESYS_CONTEXT {
-        self.esys_context.as_mut().unwrap().as_mut_ptr() // will only fail if called from Drop after .take()
-    }
-}
-
-impl Drop for Context {
-    fn drop(&mut self) {
-        info!("Closing context.");
-
-        // Flush the open handles.
-        self.open_handles.clone().iter().for_each(|handle| {
-            info!("Flushing handle {}", *handle);
-            if let Err(e) = self.flush_context(*handle) {
-                error!("Error when dropping the context: {}.", e);
-            }
-        });
-
-        let esys_context = self.esys_context.take().unwrap(); // should not fail based on how the context is initialised/used
-        let tcti_context = self.tcti_context.take().unwrap(); // should not fail based on how the context is initialised/used
-
-        // Close the TCTI context.
-        unsafe { Tss2_TctiLdr_Finalize(&mut tcti_context.into_raw()) };
-
-        // Close the context.
-        unsafe { Esys_Finalize(&mut esys_context.into_raw()) };
-        info!("Context closed.");
-    }
-}
+// #[macro_use]
+// macro_rules! wrap_buffer {
+//     ($buf:expr, $buf_type:ty, $buf_size:expr) => {{
+//         if $buf.len() > $buf_size {
+//             return Err(Error::local_error(ErrorKind::WrongParamSize));
+//         }
+//         let mut buffer = [0_u8; $buf_size];
+//         buffer[..$buf.len()].clone_from_slice(&$buf[..$buf.len()]);
+//         let mut buf_struct: $buf_type = Default::default();
+//         buf_struct.size = $buf.len().try_into().unwrap(); // should not fail since the length is checked above
+//         buf_struct.buffer = buffer;
+//         buf_struct
+//     }};
+// }
```

Comment:
If not used anymore, feel free to delete this macro!

Reply:
I will